testing/fil_token_integration/actors/basic_receiving_actor/src/lib.rs

```diff
@@ -0,0 +1,39 @@
+//use fil_fungible_token::receiver::types::TokenReceivedParams;
+use fvm_dispatch::match_method;
+use fvm_ipld_encoding::{de::DeserializeOwned, RawBytes};
+use fvm_sdk as sdk;
+use fvm_shared::error::ExitCode;
+use sdk::NO_DATA_BLOCK_ID;
+
+/// Grab the incoming parameters and convert from RawBytes to deserialized struct
+pub fn deserialize_params<O: DeserializeOwned>(params: u32) -> O {
+    let params = sdk::message::params_raw(params).unwrap().1;
+    let params = RawBytes::new(params);
+    params.deserialize().unwrap()
+}
+
+#[no_mangle]
+fn invoke(_input: u32) -> u32 {
+    let method_num = sdk::message::method_number();
+    match_method!(method_num, {
+        "Constructor" => {
+            // this is a stateless actor so constructor does nothing
+            NO_DATA_BLOCK_ID
+        },
+        "TokensReceived" => {
+            // TokensReceived is passed a TokenReceivedParams
+            //let _params: TokenReceivedParams = deserialize_params(input);
```

Comment:
This seems to explode at runtime when I build the actor with it. Not sure what's wrong yet as it should be the same type we're `send`ing over when `mint()` calls the hook. Need to look into it further.